frontend/src/container/GridGraphComponent/index.tsx

```diff
@@ -20,10 +26,21 @@ function GridGraphComponent({
 	yAxisUnit,
 	staticLine,
 	onDragSelect,
+	graphsVisibilityStates,
 }: GridGraphComponentProps): JSX.Element | null {
-	const location = history.location.pathname;
+	const { pathname } = useLocation();
 
-	const isDashboardPage = location.split('/').length === 3;
+	const isDashboardPage = pathname.split('/').length === 3;
+
+	const lineChartRef = useRef<ToggleGraphProps>();
+
+	useEffect(() => {
```

Comment:
as shared offline we need to bring the concept of PANEL_TYPES as this would be running for all types of panel

Reply:
okay